docs/encyclopedia/temporal-nexus-security.mdx

```diff
@@ -0,0 +1,55 @@
+
+To process a `ScheduleNexusOperation` command from a caller Workflow, Temporal Cloud obtains the handler Namespace and task-queue for the handler endpoint, and restricts access by verifying that the caller’s Namespace is in the endpoint’s allowlist.
+
+In this way, Temporal Cloud acts as a trusted broker across Namespace boundaries, and relies on authenticated workers in each Namespace. There is currently no provision for a handler’s endpoint to require a specific level of caller worker authentication, so either mTLS or API keys will be allowed as defined in the caller Namespace configuration. Nexus is designed to support additional Nexus access control policies in the future.
+
+#### Secure Routing {#secure-routing}
+
+In Temporal Cloud multiple security provisions are in place to ensure it can act as a trusted broker across Namespace boundaries:
+
+- Workers authenticate to their Namespaces via mTLS or an API key as allowed by their Namespace configuration.
+- mTLS is used for all Nexus communication, including across cloud cells and regions, to:
+  - Start or Cancel a `Nexus Operation`.
+  - Callback on completion of an asynchronous `Nexus Operation`.
+- Nexus `Endpoints` are only privately accessible from within a Temporal Cloud account
+  - Accessible from within a caller Workflow using the Temporal SDK
+  - Not externally accessible for arbitrary clients yet.
+- Future support for external Nexus `Endpoints` may have additional security measures such as:
+  - Signed callback URLs with single-use tokens to prevent forgery and attenuate permissions, for use over public networks.
+
+#### Payload Encryption & Data Converter {#payload-encryption-&-data-converter}
+
+The `DataConverter` works the same for a Nexus `Operation` as it does for other payloads sent between a Worker and Temporal Cloud. Currently there is support for a single `DataConverter` to be used for inbound inputs and outbound outputs, which means the encryption key often used in a custom `DataConverter` must be the same for both the caller and the handler.
+
+While there is currently no provision for using separate asymmetric encryption keys for each Namespace, we are open to adding support if required, so please reachout if you need this. This may require the exchange of public encryption keys across Namespaces, which could be inband (perhaps via the service registry) or out-of-band via existing org-specific infrastructure. Regardless, if separate asymmetric encryption keys are used the input for a Nexus `Operation` must be encrypted by the caller in a manner it can be decrypted by the handler and vice versa for the outputs, which would require changes to how the `DataConverter` works today.
```

Comment:
```suggestion
While there is currently no provision for using separate asymmetric encryption keys for each Namespace, we are open to adding support if required, so please contact us via XYZ XYZ if you need this. This may require the exchange of public encryption keys across Namespaces, which could be inband (perhaps via the service registry) or out-of-band via existing org-specific infrastructure. Regardless, if separate asymmetric encryption keys are used the input for a Nexus `Operation` must be encrypted by the caller in a manner it can be decrypted by the handler and vice versa for the outputs, which would require changes to how the `DataConverter` works today.
```

I would change this to an actual contact link now that it's going into Docs, eg like we have in https://docs.temporal.io/production-deployments/migration#multi-cluster-replication

Reply:
One line per sentence

Reply:
I wouldn't even mention reaching out to us in the public docs. And probably would just avoid mentioning unimplemented features as a general guideline.

Reply:
Agreed.